### ISSUE TITLE
Documented windows build instructions are out of date

### ISSUE THREAD

**Issue by jhseon** [spacepy/spacepy#107]
Hello, I have been trying to install spacepy on Windows10  according to the "easy way" as described in the posted installation guide on  https://pythonhosted.org/SpacePy/install_windows.html.

 I have downloaded python2.7 (python(x,y)), ffnet-0.7.1.win32-py2.7.exe, and CDF34_1_0-SETUP-32.EXE) from the links provided from the installation site. I believe I have not done anything "new"  for this installation process to avoid any unwanted or unintended mistakes, but I am still getting a library build failure for irbem (message captured in the attached file).  Spacepy setup.py recommended to try a different compiler for fortran, but then a message pops up only gnu95 is supported for Win32 machine.

 I suppose that the failure for the irbemlib on my machine is perhaps related to some incompatibility of the fortan compiler that I have downloaded from the MinGW site.... After a few trials to install spacepy on Windows10 (64-bit) without a success yet, I now wonder if I should have tried the installation on Linux machine. 

Has any body successfully installed spacepy on Windows10? I appreciate if anybody could suggest how this issue (irbem library failure, perhaps associated with fortran compiler on WIndows10) could be fixed. Any general advice is also appreciated.

I am using Python 2.7 on Windows10

 
![irbemlib_failure](https://user-images.githubusercontent.com/48402242/54082100-e384f980-4353-11e9-99d7-2435cfcc655e.png)

**Comment by drsteve** [comment]
Hi @jhseon 
Thanks for getting in touch. There are a few of us who have built SpacePy on Windows 10, but it can be an advanced move...

If you are happy working on linux then that's obviously the easiest way, but it doesn't solve the "build on windows" issue.

So, the recommendations in the SpacePy docs for building on Windows are legacy at this point. 
There are a few ways to do the install:
1. Install Miniconda or Anaconda, then make a new environment. Gfortran is available through conda (I don't recall which channel I've used in the past). Once the compilers are installed then the build should work, although I have had issues with conda setting paths to linkers incorrectly. YMMV.
2. Another conda method. Take a look in the developers/scripts directory in the repository. There are some scripts there to set up and build on Windows. I believe the paths are hard coded for miniconda3, but it's relatively straightforward to change the paths.
3. If you have admin rights then I have recently been told that linux subsystem for windows works incredibly well (formerly known as bash for windows). So you should be able to enable that, build as if it were linux, and then you'll have a functional install on Windows.

Steve

**Comment by jtniehof** [comment]
There's been a lot of work on this since the last release. I'll be able to
post instructions in a day or two and that can be the basis also for
updating the docs.

**Comment by jhseon** [comment]
Many thanks for drsteve and  jtniehof,

 The installation on Linux did not repeat the problem I had on irbem library, but I do hope that I could try an installation on Windows 10 according to the to-be-posted instructions becuase switching back-and-forth between the different OSs may not turn out be productive for some people like me...
I appreciate once again for your support..

**Comment by jtniehof** [comment]
I think the most straightforward way now is probably Anaconda. There will be a binary installer of some sort for the next release. In the meantime, after installing Anaconda:
```
conda install -y numpy scipy matplotlib networkx m2w64-gcc-fortran libpython h5py
SET FC_VENDOR=gfortran
pip install ffnet
pip install git+https://github.com/spacepy/spacepy.git
```

This will grab the latest (bleeding-edge) version from the repository and install it. If you're willing to try it out, I'd be very interested in hearing of any difficulties, since that would help us fix before release.

**Comment by jhseon** [comment]
Thanks to the suggestions, the symptom is now gone and it seems that the necessary libraries are recognized (especially irbem). 

 I have listed all the steps, just in case someone else is interested in the installation on Windows 10. FYI,  I have added git and CDF install commands as it seems to be needed.

I appreciate your support again and  will report if there are any further issues.

OS:  Windows: Windows 10 (64 bit),   Build 17134.619 (Clean start after C formatting)

step | activity

1 | Install Anaconda 3.7 (64 bit) from  official site (https://www.anaconda.com/distribution/)

2 | run "conda install -y numpy scipy   matplotlib networkx m2w64-gcc-fortran libpython h5py"

   Installed version --> (numpy=1.15.4, matplotlib=3.0.2, scipy=1.1.0)

3 | run "SET FC_VENDOR=gfortran"

4 | run "pip install ffnet"

5 | run "conda install git"

6 | Install CDF from official site   (https://spdf.sci.gsfc.nasa.gov/pub/software/cdf/dist/cdf37_0/windows/, 64 bit version)

7 | pip install   git+https://github.com/spacepy/spacepy.git

**Comment by jtniehof** [comment]
Note because of #160 / #161 when updating the Windows instructions we should include the ffnet information.

**Comment by jtniehof** [comment]
This was largely fixed in e43b88f11d9db6bd32134594d460cf529c2397df (with refinements later, e.g. 28c85800a251ba22b759665528dc025519789581 and c70efddb05dacc07a7754974ba568d776d6c15d8) so I'm closing this. I'm sure there are always improvements to be made to the Windows install docs but they're no longer completely obsolete.
